underscore.js

```diff
@@ -877,6 +877,23 @@
     };
   };
 
+  // Returns the first non-null, defined value
```

Comment:
Unless that value is a function, in which case it invokes it a few times and checks the result instead. That's some surprising behaviour right there. What you've done is lose parametricity. A nice `coalesce` function would treat all values the same -- an array of strings, an array of numbers, and, yes, an array of functions. This is why everyone hates `Array.prototype.concat`: concat creates a new array with the given value on the end *unless* that value is an array (technically @@isConcatSpreadable as of ES6), in which case it will instead have each of the array's values on the end. I would write this function like this:

```js
_.coalesce = _.partial(_.find, _, x => x != null)
```